waspc/packages/deploy/src/providers/fly/helpers/helpers.ts

```diff
@@ -1,127 +1,22 @@
-	waspSays('Please double check your path.');
-	exit(1);
-}
+import {
+    ensureDirAbsoluteAndExists,
+    getWaspBuildDir,
+} from '../../shared/helpers.js';
 
 export function buildDirExists(waspProjectDir: string): boolean {
-	const waspBuildDir = getWaspBuildDir(waspProjectDir);
-	return fs.existsSync(waspBuildDir);
-}
-
-export function cdToServerBuildDir(waspProjectDir: string): void {
-	const waspBuildDir = getWaspBuildDir(waspProjectDir);
-	cd(waspBuildDir);
-}
-
-export function cdToClientBuildDir(waspProjectDir: string): void {
-	const waspBuildDir = getWaspBuildDir(waspProjectDir);
-	cd(path.join(waspBuildDir, 'web-app'));
-}
-
-function getWaspBuildDir(waspProjectDir: string) {
-	return path.join(waspProjectDir, '.wasp', 'build');
-}
-
-export function ensureDirsInCmdAreAbsoluteAndPresent(thisCommand: Command): void {
-	const waspProjectDirPath: string | undefined = thisCommand.opts().waspProjectDir;
-	if (waspProjectDirPath) {
-		if (!path.isAbsolute(waspProjectDirPath)) {
-			waspSays('The Wasp dir path must be absolute.');
-			exit(1);
-		}
-
-		const waspProjectDirExists = fs.existsSync(waspProjectDirPath);
-		if (!waspProjectDirExists) {
-			waspSays('The Wasp dir path does not exist.');
-			exit(1);
-		}
-	}
-
-	const flyTomlDirPath: string | undefined = thisCommand.opts().flyTomlDir;
-	if (flyTomlDirPath) {
-		if (!path.isAbsolute(flyTomlDirPath)) {
-			waspSays('The toml dir path must be absolute.');
-			exit(1);
-		}
-
-		const flyTomlDirExists = fs.existsSync(flyTomlDirPath);
-		if (!flyTomlDirExists) {
-			waspSays('The toml dir path does not exist.');
-			exit(1);
-		}
-	}
-}
-
-// eslint-disable-next-line
-export function makeIdempotent<F extends () => any>(
-	fn: F,
-): () => ReturnType<F> {
-	let result: { value: ReturnType<F> } | null = null;
-
-	return function idempotentFn() {
-		if (!result) {
-			result = { value: fn() };
-		}
-		return result.value;
-	};
-}
-
-// For some reason, the colors from the chalk package wouldn't
-// show up when run as a subprocess by the Wasp CLI. This works.
-export function waspSays(str: string): void {
-	console.log('🚀 \x1b[33m ' + str + ' \x1b[0m');
+    const waspBuildDir = getWaspBuildDir(waspProjectDir);
+    return fs.existsSync(waspBuildDir);
 }
 
-export function displayWaspRocketImage(): void {
-	// Escaping backslashes makes it look weird here, but it works in console.
-	const asciiArt = `
-
-                    __
-                   // \\
-                   \\\\_/ //
-             _    -(||)(')
-            \\ \\____///_____
-   #########[==__DEPLOYED__}
-            /_/
-
-  `;
-	console.log(asciiArt);
-}
-
-export function getCommandHelp(command: Command): string {
-	return trimUsage(command.helpInformation());
-}
-
-function trimUsage(usage: string): string {
-	return usage.split(/[\r\n]+/)[0].replace('Usage: ', '').replace(' [options]', '');
-}
+export function ensureDirsInCmdAreAbsoluteAndPresent(
```

Comment:
Can this be a shared hook?

Reply:
I think I tried, but `fly` checks for different folder, like `toml dir` while `Railway` doesn't at all, apart from `wasp dir`. So while they need to do similar types of checks (existence of dir), dirs are different. So extracted dir checking logic and just making each provider check different dirs on its own. But it could be probably done a bit different, if we'd set a getter for dirs in Command extension, and then in the shared util just use this getter to verify dirs. Only thought of this now

Reply:
Ah ok, sorry, I totally missed the Fly toml check in the diff. Separate seems fine to me, thanks for the explanation! 👍🏻